### PR TITLE
feat(in-app-messaging): merge feature to next-release/main

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Merge In-App Messaging feature to `next-release/main`

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changeset/green-eels-sniff.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"next-example": patch
```
Comment (reesscot) [inline .changeset/green-eels-sniff.md:2]: Is this changeset necessary?

File: .github/changeset-presets/bump-react.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 '@aws-amplify/ui': patch
 '@aws-amplify/ui-react': patch
 '@aws-amplify/ui-react-core': patch
+'@aws-amplify/ui-react-native': patch
```
Comment (reesscot) [inline .github/changeset-presets/bump-react.md:5]: InAppMessaging is worth of a minor bump, it's a feature! ⚡️💪 ⚡️

**Review by reesscot — DISMISSED** [review]

File: packages/react-native/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+## Features 🚀
+
+- **Better developer experience** Connected-components like Authenticator are being written with framework-specific implementations so that they follow framework conventions and are easier to integrate into your application.
+- **Endlessly customizable** Every detail of Amplify UI is customizable to match your brand. Style all of Amplify UI with themes, override components with your own, or build your own UI and use Amplify for complex state management.
+- **Accessible** Amplify UI components follow [WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/) and [WAI-ARIA](https://www.w3.org/TR/wai-aria-1.2/) best practices and guidelines such as color contrast, keyboard navigation, accessible labels, and focus management.
+- **Primitive components (React only right now)** Primitive components are used in the connected components, like Authenticator, you can also customize them and use them to build the rest of your UI.
+
+## We love contributors!!
+
+See our contributing guide [CONTRIBUTING.md](/CONTRIBUTING.md) to help us scale Amplify UI!
```
Comment (wlee221) [inline packages/react-native/README.md:23]: for later, it'd be nice to have a `CONTRIBUTING.md` for `react-native` like [how we do](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md) for `react`.

File: packages/react-native/package.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "@aws-amplify/ui-react-native",
+  "version": "0.0.1",
```
Comment (wlee221) [inline packages/react-native/package.json:3]: 1.0.0 instead of 0.0.1?

File: packages/react/package.json [hunk 5]
```diff
@@ -67,6 +67,7 @@
     "qrcode": "1.5.0",
     "react-generate-context": "1.0.1",
     "react-map-gl": "7.0.15",
+    "tinycolor2": "^1.4.2",
```
Comment (wlee221) [inline packages/react/package.json:70]: Can we pin the dependency to prevent upstream changes from bleeding in?

File: packages/ui/package.json [hunk 6]
```diff
@@ -8,6 +8,7 @@
       "import": "./dist/esm/index.js",
       "require": "./dist/index.js"
     },
+    "./package.json": "./package.json",
```
Comment (wlee221) [inline packages/ui/package.json:11]: Why was this added?

**Review by wlee221 — COMMENTED** [review]
LGTM, just few questions

File: packages/ui/package.json (hunk 6 above)
Comment (calebpollman) [reply packages/ui/package.json:11]: It's a requirement for Metro bundler (React Native bundler)

**Review by wlee221 — APPROVED** [review]

File: .github/changeset-presets/bump-react.md [hunk 7]
```diff
@@ -2,6 +2,7 @@
 '@aws-amplify/ui': patch
 '@aws-amplify/ui-react': patch
```
Comment (wlee221) [inline .github/changeset-presets/bump-react.md:3]:
```suggestion
'@aws-amplify/ui-react': minor
```
since it's a new feature

**Review by reesscot — DISMISSED** [review]

**Review by wlee221 — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
